### ISSUE TITLE
Disallow overloading of parsers, controls, packages

### ISSUE THREAD

**Issue by mihaibudiu** [p4lang/p4-spec#1080]
## Personnel
- [x] **Owner**: @mbudiu-vmw 
- [x] **Supporters**: @apinski-cavium 

## Design
- [x] **Document**: See below

## Implementation
- [ ] **`p4-spec`**:
- [ ] **`p4c`**:
## Process
- [ ] **LDWG discussed**:
- [ ] **LDWG approved**:
- [ ] **Merged into `p4-spec`**:
- [ ] **Merged into `p4c`**:

=======================================

**Comment by mbudiu-vmw** [comment]
See https://github.com/p4lang/p4-spec/pull/1078
Overloading is subtle, since it can be made based on both constructor and apply arguments.

**Comment by mbudiu-vmw** [comment]
Consider this example from @apinski-cavium 
```
Take:
parser p() () {
  state start {
    transition accept;
  }
}
parser p() (bit a) {
  state start {
    transition accept;
  }
}
parser p(bool b) (bit a) {
  state start {
    transition accept;
  }
}
parser p(bool a) () {
  state start {
    transition accept;
  }
}


parser p1() () {
  p() a;
  state start {
    a();
    transition accept;
  }
}
```

**Comment by jnfoster** [comment]
Closing as it seems this has been resolved in #1078.
